### PR TITLE
Use metomi-isodatetime in preference to isodatetime

### PR DESCRIPTION
**Pull request by wxtim**
## Change dependency in Rose from isodatetime to metomi.isodatetime
The isodatetime module was recently renamed metomi.isodatetime ([PR 132](https://github.com/metomi/isodatetime/pull/132)). 

This PR changes the namespaces in cylc to reflect this.

- [x] Grepping for `isodatime` without `metomi` produces reasonable results.
- [x] All existing tests pass - this is unlikely to happen before [Cylc-flow PR3222](https://github.com/cylc/cylc-flow/pull/3222) is accepted.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -20,7 +20,7 @@ jobs:
   include:
     - name: "Unit Tests"
       install:
-        - now install coverage linters pytest cylc rose
+        - now install coverage linters pytest isodatetime cylc rose
```
Comment (oliver-sanders) [inline .travis.yml]: Shouldn't have to install it here as the dependency is already in the `setup.py`.

**Comment by wxtim** [comment]
@sadielbartholomew @oliver-sanders This review can be re-activated now Isodatetime is on PyPI

**Comment by oliver-sanders** [comment]
Looks OK.

Note to reviewers: Travis installes isodatetime:master (which is probably the way to go).

We can remove stuff which was related to the bundling of isodatetime:

* .codacy.yml
* .codecov.yml
* .coveragerc
* ACKNOWLEDGEMENT.md
* tox.ini

**Comment by oliver-sanders** [comment]
Probably best to get the Cylc PR merged before merging this one to make sure nothing breaks.

**Comment by oliver-sanders** [comment]
To get this PR to behave like https://github.com/cylc/cylc-flow/pull/3222 you will need to remove the `_install_isodatetime` logic from `.travis/now`.

File: .travis/now [hunk 2]
```diff
@@ -92,9 +92,7 @@ _install_cylc () {
 }
 
 _install_isodatetime () {
-    wget 'https://github.com/metomi/isodatetime/archive/master.tar.gz' \
-        -O - | tar -xz -C "${HOME}"
-    pip install -e "${HOME}/isodatetime-master"
+    pip install metomi-isodatetime
```
Comment (oliver-sanders) [inline .travis/now]: This can be removed all-together as metomi-isodatetime is a dependency of Rose.

**Comment by wxtim** [comment]
sorry- can I poke @sadielbartholomew  and @oliver-sanders  for a re-review.

**Comment by oliver-sanders** [comment]
Jinja2 site is down which is causing documentation test failure, this can be ignored.

**Pull request merged**